wait.go

```diff
@@ -242,11 +242,11 @@ func (c *Client) IsReady(item *unstructured.Unstructured) (bool, string) {
 	}
 	for _, raw := range conditions {
 		condition := raw.(map[string]interface{})
-		if condition["type"] != "Ready" && condition["status"] != "True" {
-			return false, fmt.Sprintf("⏳ waiting for %s/%s: %s", condition["type"], condition["status"], condition["message"])
+		if condition["type"] == "Ready" && condition["status"] == "True" {
```

Comment:
This changes the logic from wait for all non "Ready" conditions, to wait on *only" Ready condition - It also drops the detail on which condition it is waiting on.

I think we should by convention assume conditions have a negative polarity (Invalid, DiskPressure, etc..) and then have a separate list for positive conditions e.g.  Ready

For a resource to be ready it should be:
* use custom IsReady if available
* if non core type, len(conditions) > 0 && all(positive-conditions)  == true && all(negative-conditions) == false
* if a data container (configmap/secret/endpoint) - non empty
* if a service type load balancer - has a loadBalancerIp
* if another service type - has endpoints
* for  rbac.authorization.k8s.io - always ready
* for PVC = status.Phase = Bound

wdyt @teodor-pripoae @Kaitou786 ?

Reply:
I agree with your conclusion, just for the case `another service type` I personally think that it should be always ready. For example, an error in setting up the label could cause the endpoints to never appear... does that mean that the object is not ready?
and IIRC you don't see the endpoints in the `yaml/json` object at least when you try `k get svc xyz -o yaml`

Reply:
@moshloop I agree with you. I didn't know the original intention of this `WaitForResource` so I made a wild guess of how it should be (strictly checking the "Ready" conditions for any unknown resource type). But it makes sense with your explanation now.

I think we should implement `IsReady` for everything that we know of (including all the things you listed) and try to avoid wildcard/general checking as much as possible. And for resources that don't have `IsReady`, it should be marked as ready as long as the resource exists. WDYT?

@Kaitou786 IMHO, if the selector in the service is incorrectly set up (thus endpoint is not created), we can say that the service is not ready. I think that makes sense because it helps us catch errors in an incorrect Service definition.

Reply:
@tobernguyen  We need to support some type of duck-typing as this function is used by the template operator to determine readiness on custom templates which can include resources we don't yet know about, but still follow conditions conventions